webp/view/src/main/cpp/webp_decode.cpp

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (C) The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cassert>
+#include <pthread.h>
+#include <webp/decode.h>
+#include "webp_decode.h"
+
+WebpDecoder::WebpDecoder(const char** files, uint32_t count,
+                         DecodeSurfaceDescriptor* frameBuf,
+                         AAssetManager* assetMgr)
+    : assetMgr_(assetMgr),
+      buf_(nullptr), stopPending_(false),
+      state_(state_idle), worker_(nullptr) {
+    for (auto i = 0; i < count; i++)
+        files_.push(files[i]);
```

Comment:
We encourage putting braces all the time to avoid goto fail; bug.
https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/

for (auto i = 0; i < count; i++) {
}